tests/tests_git/test_git_repository.py

```diff
@@ -57,6 +57,27 @@ def test_output(self, mock_subprocess: Any) -> None:
         self.repo.output("echo hello")
         mock_subprocess.assert_called_with("echo hello", cwd=self.repo.dir, shell=True)
 
+    @patch("subprocess.check_output", return_value='927de30 2022-07-12'.encode())
+    def test_log(self, mock_subprocess: Any) -> None:
+        log = self.repo.log('2021-10-26')
+        last_commit = log[-1]
+        self.assertEqual(last_commit.id, "927de30")
+        self.assertEqual(last_commit.date, "2022-07-12")
+
+    @patch("subprocess.check_output", return_value=''.encode())
+    def test_log_empty(self, mock_subprocess: Any) -> None:
+        log = self.repo.log('2021-10-26')
+        last_commit = log[-1]
+        self.assertEqual(last_commit.id, None)
+        self.assertEqual(last_commit.date, None)
+
+    @patch("subprocess.check_output", return_value='927de30 2022-07-12\n787d971 2022-04-28\n989143e 22-04-26\n8b376fc 2022-04-26\n023a2ac 2022-04-26'.encode())
+    def test_log_multiple(self, mock_subprocess: Any) -> None:
+        log = self.repo.log('2022-04-25')
+        last_commit = log[-1]
```

Comment:
check the number of entries returned from `.log` match your expectation (I see 5 items)

Reply:
Modifies this to add more assertions that tests `gitlog` with one-commit, no-commit and more-commits.